osquery/main/windows/main.cpp

```diff
@@ -342,35 +287,29 @@ Status uninstallService() {
 void WINAPI ServiceControlHandler(DWORD control_code) {
   switch (control_code) {
   case SERVICE_CONTROL_STOP:
-  case SERVICE_CONTROL_SHUTDOWN:
+  case SERVICE_CONTROL_SHUTDOWN: {
     if (kServiceStatus.dwCurrentState != SERVICE_RUNNING) {
       break;
     }
 
     // Give the main thread a chance to shutdown gracefully before exiting
     UpdateServiceStatus(0, SERVICE_STOP_PENDING, 0, 3, kServiceShutdownWait);
-    {
-      auto stopEvent = osquery::getStopEvent();
-      if (stopEvent != nullptr) {
-        auto ret = SetEvent(stopEvent);
-        if (ret != TRUE) {
-          SLOG("SetEvent failed (lasterror=" + std::to_string(GetLastError()) +
-               ")");
-        }
-        CloseHandle(stopEvent);
-      }
-      auto thread = OpenThread(SYNCHRONIZE, false, kLegacyThreadId);
-      if (thread != nullptr) {
-        WaitForSingleObjectEx(thread, INFINITE, FALSE);
-        CloseHandle(thread);
-      } else {
-        SLOG("Failed to open handle to main thread of execution with " +
-             std::to_string(GetLastError()));
-      }
+
+    Initializer::requestShutdown();
+    auto thread = OpenThread(SYNCHRONIZE, false, kLegacyThreadId);
+    if (thread != nullptr) {
+      WaitForSingleObjectEx(thread, INFINITE, FALSE);
+      CloseHandle(thread);
+    } else {
+      auto le = GetLastError();
+      SLOG("Failed to open handle to main thread of execution with " +
+           std::to_string(le));
     }
+
     UpdateServiceStatus(0, SERVICE_STOPPED, 0, 4);
```

Comment:
Found out why it errors out on Windows, this shouldn't be done here, since the ServiceMain it's already (correctly) doing it.